core/src/testils/criterion.rs

```diff
@@ -25,12 +30,26 @@ use std::cmp;
 use std::sync::mpsc::{self, Receiver};
 use std::time::{Duration, Instant};
 
+/// Criterion `Bencher` extension trait.
 pub trait BencherExt {
+    /// Times a `routine` with an input generated via a `proptest strategy`
+    /// batch of input, and then times the iteration of the benchmark over the
+    /// input. See [`BatchSize`] for details on choosing the batch size. Use
+    /// this when the routine must consume its input.
```

Comment:
last sentence can be "The `routine` consumes its input." since we don't have an alternative option here.